Lib/typing.py

```diff
@@ -1941,6 +1967,29 @@ class Film(TypedDict):
     return isinstance(tp, _TypedDictMeta)
 
 
+def assert_never(arg: Never, /) -> Never:
+    """Statically assert that a line of code is unreachable.
+
+    Example::
+
+        def int_or_str(arg: int | str) -> None:
+            match arg:
+                case int():
+                    print("It's an int")
+                case str():
+                    print("It's a str")
+                case _:
+                    assert_never(arg)
+
+    If a type checker finds that a call to assert_never() is
+    reachable, it will emit an error.
+
+    At runtime, this throws an exception when called.
+
+    """
+    raise RuntimeError("Unreachable code")
```

Comment:
Maybe use a message like "Expected code to be unreachable" ? Clearly it's not unreachable.

(Aside: This kind of runtime error I expect to occur if someone writes code that is NOT actually unreachable and hasn't yet run a typechecker on it, which would complain.)

Reply:
Yes, that's better, thanks. I'll update the PR tomorrow.